pkg/controller/operconfig/ingressconfig_controller.go

```diff
@@ -0,0 +1,115 @@
+		log.Printf("Reconciling update to %s/%s\n", request.Namespace, request.Name)
+		ingressControllerConfig := &operv1.IngressController{TypeMeta: metav1.TypeMeta{APIVersion: operv1.GroupVersion.String(), Kind: "IngressController"}}
+		err := r.client.Get(context.TODO(), request.NamespacedName, ingressControllerConfig)
+		if err != nil {
+			if apierrors.IsNotFound(err) {
+				log.Printf("Ingress Controller configuration %s was deleted", request.NamespacedName.String())
+				// Request object not found, could have been deleted after reconcile request.
+				// Owned objects are automatically garbage collected, since we set
+				// the ownerReference (see https://kubernetes.io/docs/concepts/workloads/controllers/garbage-collection/).
+				// Return and don't requeue
+				return reconcile.Result{}, nil
+			}
+			// Error reading the object - requeue the request.
+			log.Printf("Unable to retrieve IngressController.operator.openshift.io object: %v", err)
+			return reconcile.Result{}, err
+		}
+		labels := map[string]string{HostNetworkLabel: ""}
+		log.Printf("Ingress Controller Config: %+v", ingressControllerConfig)
+		if ingressControllerConfig.Spec.EndpointPublishingStrategy != nil &&
+			ingressControllerConfig.Spec.EndpointPublishingStrategy.Type == operv1.HostNetworkStrategyType {
+			labels[HostNetworkLabel] = HostNetworkLabelValue
+		}
+		err = r.setLabelsOnNamespace(HostNetworkNamespace, labels)
+		if err != nil {
+			log.Printf("Error setting the host network label on namespace %s: %v", HostNetworkNamespace, err)
+			return reconcile.Result{}, err
+		}
+		return reconcile.Result{RequeueAfter: ResyncPeriod}, nil
+	}
+	return reconcile.Result{}, nil
+}
+
+// setLabelsOnNamespace sets the labels specified on the target namespace using the client API
+func (r *ReconcileIngressConfigs) setLabelsOnNamespace(targetNamespace string, labels map[string]string) error {
+	var err error
+	namespace := &corev1.Namespace{TypeMeta: metav1.TypeMeta{APIVersion: corev1.SchemeGroupVersion.String(), Kind: "Namespace"}}
+	err = r.client.Get(context.TODO(), types.NamespacedName{Name: targetNamespace}, namespace)
+	if err != nil {
+		if apierrors.IsNotFound(err) {
+			log.Printf("Namespace %s was deleted", targetNamespace)
+			// Request object not found, could have been deleted after reconcile request.
+			// Owned objects are automatically garbage collected, since we set
+			// the ownerReference (see https://kubernetes.io/docs/concepts/workloads/controllers/garbage-collection/).
+			// Return and don't requeue
+			return nil
+		}
+		// Error reading the object - requeue the request.
+		log.Printf("Unable to retrieve namespace object: %v", err)
+		return err
+	}
+	newNamespace := namespace.DeepCopy()
+	existingLabels := newNamespace.GetLabels()
+	if existingLabels == nil {
+		existingLabels = map[string]string{}
+	}
+	for k, v := range labels {
+		if v == "" {
+			delete(existingLabels, k)
```

Comment:
this is kind of weird semantics given that `""` is a very commonly-used value for labels/annotations.

You could just pass a bool value into `setLabelsOnNamespace` rather than a map, and have it know what to do with that. That would actually simplify the code quite a bit anyway.

Reply:
or a `*string` to indicate a deleted label.